networks/rpc/http_newrelic.go

```diff
@@ -0,0 +1,136 @@
+package rpc
+
+import (
+	"bytes"
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"runtime/debug"
+
+	"github.com/gin-gonic/gin/binding"
+	"github.com/newrelic/go-agent/v3/newrelic"
+)
+
+// dupWriter writes data to the buffer as well as http response
+type dupWriter struct {
+	http.ResponseWriter
+	body *bytes.Buffer
+}
+
+func (w dupWriter) Write(b []byte) (int, error) {
+	w.body.Write(b)
+	return w.ResponseWriter.Write(b)
+}
+
+// KASAttrs contains identifications for a KAS request
+type KASAttrs struct {
+	ChainID      string `header:"x-chain-id" json:"x-chain-id"`
+	AccountID    string `header:"x-account-id" json:"x-account-id"`
+	RequestID    string `header:"x-request-id" json:"x-request-id"`
+	ParentSpanID string `header:"x-b3-parentspanid,omitempty" json:"x-b3-parentspanid,omitempty"`
+	SpanID       string `header:"x-b3-spanid,omitempty" json:"x-b3-spanid,omitempty"`
+	TraceID      string `header:"x-b3-traceid,omitempty" json:"x-b3-traceid,omitempty"`
+}
+
+func parseKASHeader(r *http.Request) KASAttrs {
+	attrs := KASAttrs{}
+	if err := binding.Header.Bind(r, &attrs); err != nil {
+		logger.Error("failed to bind a KAS header")
+	}
+	return attrs
+}
+
+// newNewRelicHTTPHandler enables NewRelic web transaction monitor.
+// It also prints error logs when RPC returns contains error messages.
+func newNewRelicHTTPHandler(nrApp *newrelic.Application, handler http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		defer func() {
+			if err := recover(); err != nil {
+				logger.Error("NewRelic http handler panic\n"+string(debug.Stack()), "err", err)
```

Comment:
How about using `ErrorWithStack`?
```suggestion
				logger.ErrorWithStack("NewRelic http handler panic", "err", err)
```

Reply:
`ErrorWithStack` is same with below code from `serverRequest`?
```
			const size = 64 << 10
			buf := make([]byte, size)
			buf = buf[:runtime.Stack(buf, false)]
			logger.Error(string(buf))
```

Reply:
@KimKyungup You're right. The same action except for minor configurations. And, this code will handle the panic raised on the handler level code (before `serverRequest`).